### PR TITLE
Feature - 2921 - Talent Search Login/Logout/Register Links

### LINKED ISSUE #2921: Add login and register links to talentsearch nav bar (no modals)
Our shared navbar on talentseach needs links to the new login and register pages.

[
image
](https://www.figma.com/file/UDWAkiGXMt9YGbrfsuSzmr/Interface---Registration-and-login?node-id=6%3A2)

## User Acceptance Critera
- [ ] Can navigate to /[lang]/login-info
- [ ] Can navigate to /[lang]/register

### PR DESCRIPTION
**Pull request by esizer**
Resolves #2921 

## Summary

This changes the `@common/components/NavMenu` to accept both `mainItems` and `utilityItems`. The utility items will be placed on the right side if provided. In the `talentsearch` app, we have added Login, Logout and Register.

## Screenshot

Screen Shot 2022-06-08 at 10 48 00 AM

### CODE REVIEW

**Comment by vd1992** [comment]
playing around, seems to be working
upon logging on it takes you to /talent/search, which I assume will be changed after PR #2992 ?

File: frontend/talentsearch/src/js/components/PageContainer.tsx [hunk 1]
```diff
@@ -21,32 +21,42 @@ interface MenuLinkProps {
...
+  as = Link,
   isActive = exactMatch,
+  ...rest
 }) => {
   const location = useLocation();
+  const El = as;
   return (
-    <Link
+    <El
       href={href}
```
Comment (yonikid15) [inline frontend/talentsearch/src/js/components/PageContainer.tsx:41]: That's neat! Injecting the html element using a prop. However, I don't think buttons can have a `href` attribute.. can they?

**Review by yonikid15 — COMMENTED** [review]
Nice work Eric! Just had a couple questions for you or whoever can answer them.

File: .bundlewatch.config.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ module.exports = {
   files: [
     {
       path: "frontend/talentsearch/dist/app.js",
-      maxSize: "250 kB",
+      maxSize: "255 kB",
```
[the remark this answers is not in the document]
Comment (esizer) [reply .bundlewatch.config.js:14]: Yeah, my understanding is that small increases are fine. This is more there so we know when the bundle is increasing and can be aware of it.

File: frontend/talentsearch/src/js/components/PageContainer.tsx (hunk 1 above)
Comment (esizer) [reply frontend/talentsearch/src/js/components/PageContainer.tsx:42]: You are very right, thanks for catching that 🎉 . I have fixed that along with not rendering empty titles with [remove href on menu buttons](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3009/commits/c20848fec708547251b52795b14554ea2b15cfbd)

**Review by yonikid15 — APPROVED** [review]
LGTM 🎉

**Pull request merged**